x/stakeibc/keeper/msg_server_redeem_stake.go

```diff
@@ -120,10 +120,14 @@ func (k msgServer) RedeemStake(goCtx context.Context, msg *types.MsgRedeemStake)
 	// Set the UserUnbondingRecords on the proper HostZoneUnbondingRecord
 	hostZoneUnbondings := epochUnbondingRecord.GetHostZoneUnbondings()
 	if len(hostZoneUnbondings) == 0 {
-		hostZoneUnbondings = make(map[string]*recordstypes.HostZoneUnbonding)
+		hostZoneUnbondings = []*recordstypes.HostZoneUnbonding{}
 		epochUnbondingRecord.HostZoneUnbondings = hostZoneUnbondings
 	}
-	epochUnbondingRecord.HostZoneUnbondings[hostZone.ChainId] = hostZoneUnbonding
+	didSet := k.RecordsKeeper.SetHostZoneOnEpochUnbondingRecord(ctx, epochUnbondingRecord.EpochNumber, hostZone.ChainId, hostZoneUnbonding)
+	if !didSet {
+		k.Logger(ctx).Error(fmt.Sprintf("Failed to set host zone epoch unbonding record %v", err))
+		return nil, sdkerrors.Wrapf(types.ErrInsufficientFunds, "couldn't set host zone epoch unbonding record. err: %s", err.Error())
+	}
```

Comment:
remove `err` and `err.Error()` - it's referencing an `err` from earlier in the function and has potential to throw a null pointer exception. I'd maybe add the chain Id to the log as well

Reply:
good catch! done